odoo/tools/translate.py

```diff
@@ -443,8 +443,23 @@ def _get_lang(self, frame):
                     lang = env['res.users'].context_get()['lang']
         return lang
 
-    def __call__(self, source):
-        return self._get_translation(source)
+    def __call__(self, source, *args, **kw):
+        translation = self._get_translation(source)
+        if args:
+            try:
+                return translation % args
+            except (TypeError, ValueError):
```

Comment:
Under what condition does this raise a ValueError?

Reply:
```
>>> "Foo s%" % 42
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: incomplete format
```

Reply:
Same error with a mapping, so really the only difference errors-wise is the mapping one can also raise a `KeyError`, which it's fine to catch in either case.